templates/discover.html

```diff
@@ -25,7 +25,34 @@ <h2>Discover snaps for any Linux device</h2>
     </div>
   </section>
 
-  <!--
-    {{ featured_snaps }}
-  -->
+  <section class="p-strip is-deep is-bordered">
+    <div class="row">
+      <div class="col-12">
+        <h2>Featured snaps</h2>
+      </div>
+    </div>
+    <div class="row">
+      {% for snap in featured_snaps %}
+        <div class="col-3">
+          <div class="p-heading-icon">
+            <a class="p-heading-icon__header" href="/{{ snap.package_name }}">
+              {% if snap.icon_url %}
+                <img class="p-heading-icon__img" src="{{ snap.icon_url }}" alt="">
+              {% else %}
+                <img class="p-heading-icon__img" src="https://assets.ubuntu.com/v1/610689b1-default-package-icon.svg" alt="">
+              {% endif %}
+              <h5 class="u-no-margin">{{ snap.title }}</h5>
+            </a>
+          </div>
+        </div>
+
+        {# end the row after every 4th snap #}
+        {% if (loop.index % 4 == 0) and not loop.last %}
```

Comment:
Because of how Vanilla grid works we need to split the snaps into rows.
This means we need this tricky bit inside to loop to detect when the row should end and 'inject' a new row in between the snaps.

Reply:
@bartaz there's actually a very handy column() mixin - https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/grid/shelves/mixins/_column.scss  that is accessible in vanilla - it allows you to clear first column of the next row, so you don't have to generate additional .row markup. This will clear 
`        >:nth-child(3n+1) {
          @include column(6, $grid-columns, true);
        }
`
I've used it to fix a hard-coded 50% instead of 6-col here so it should be permissable: https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_patterns_lists.scss#L218

(apologies if I'm saying something you know and avoid doing because of other reasons)

Reply:
@lyubomir-popov 
I generally avoid adding CSS as we are aiming to use Vanilla as it is. Especially reusing Vanilla internal implementation (such as column() mixin and $grid-columns doesn't seem good from maintenance point of view, as these can change in Vanilla at any time).

Reply:
@bartaz Ah I see. Makes sense. Would it be possible to have the snaps display 2 per row on mobile & tablet resolutions with the above markup?

Reply:
@lyubomir-popov No. It uses grid, so on mobile it just goes down to one column.

As you are redesigning it anyway for next phase, I would propose leaving it as it is now and address these issues when implementing new design. At that point we could use similar flexbox solution to one we used for logos on home page.